// Copyright 2015 The Chromium OS Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
syntax = "proto2";
option optimize_for = LITE_RUNTIME;

import "soma_container_spec.proto";

package germ;

service Germ {
  rpc Launch (LaunchRequest) returns (LaunchResponse);
  rpc Terminate (TerminateRequest) returns (TerminateResponse);
}

message LaunchRequest {
  optional string name = 1;
  optional soma.ContainerSpec spec = 2;
}

message LaunchResponse {
  optional bool success = 1;
  // |pid| will be valid if |success| is true.
  optional int32 pid = 2;
}

message TerminateRequest {
  optional int32 pid = 1;
}

message TerminateResponse {
  optional bool success = 1;
}
